src/adapterService/edgeAdapterService.ts

```diff
@@ -290,7 +348,56 @@ export module EdgeAdapter {
 
         private createGuid(): string {
             const g: string = crypto.createHash('md5').update(Math.random().toString()).digest('hex').toUpperCase();
-            return `${g.substring(0, 8)}-${g.substring(9, 13)}-${g.substring(13, 17)}-${g.substring(17, 21)}-${g.substring(21, 31)}`
+            return `${g.substring(0, 8)}-${g.substring(9, 13)}-${g.substring(13, 17)}-${g.substring(17, 21)}-${g.substring(21, 31)}`            
+        }
+
+        private extractParametersFromUrl(url: string): {url:string, paramChain: string}{
+            let parameters;
+            const closeUrl = '/json/close';
+
+            if(url.indexOf('/json/new') != -1){
+                const urlSegements = url.split('?');                
+                if(urlSegements.length > 1){
+                    url = urlSegements[0];
+                    parameters = urlSegements[1];
+                }
+            } else if(url.indexOf(closeUrl) != -1){
+                parameters = url.replace(`${closeUrl}/`,'');
+                url = url.slice(0,closeUrl.length);
+                }
+
+            return {url: url, paramChain: parameters};
+        }
+
+        private closeEdgeInstance(guid: string): boolean{
+            var result = false;
+
+            const id = this._guidToIdMap.get(guid.toLocaleUpperCase());            
+            if(id){                
+                result = edgeAdapter.closeEdge(id);
```

Comment:
I agree, now the cleaning of the other resources is not dependent on the result of closing Edge.